packages/amplify-e2e-tests/src/rds-v2-test-utils/index.ts

```diff
@@ -158,16 +192,29 @@ const getMappedName = (definition: ObjectTypeDefinitionNode | FieldDefinitionNod
   return mappedName;
 };
 
-const getFieldStatement = (field: FieldDefinitionNode, isPrimaryKey: boolean) => {
+const getFieldStatement = (field: FieldDefinitionNode, isPrimaryKey: boolean, engine: ImportedRDSType): string => {
   const fieldName = getMappedName(field);
   const fieldType = field.type;
   const isNonNull = fieldType.kind === Kind.NON_NULL_TYPE;
   const baseType = getBaseType(fieldType);
-  const columnType = isArrayOrObject(fieldType, []) ? 'JSON' : convertToSQLType(baseType);
-  const sql = `${fieldName} ${columnType} ${isNonNull ? 'NOT NULL' : ''} ${isPrimaryKey ? 'PRIMARY KEY' : ''}`;
+  const columnType = isArrayOrObject(fieldType, []) ? getArrayStringFieldType(engine) : convertToSQLType(baseType);
+  const sql = `${convertToDBSpecificName(fieldName, engine)} ${columnType} ${isNonNull ? 'NOT NULL' : ''} ${
+    isPrimaryKey ? 'PRIMARY KEY' : ''
+  }`;
   return sql;
 };
 
+const getArrayStringFieldType = (engine: ImportedRDSType): string => {
+  switch (engine) {
+    case ImportedRDSType.MYSQL:
+      return 'JSON'; // MySQL does not support array types
+    case ImportedRDSType.POSTGRESQL:
+      return 'VARCHAR[]';
+    default:
+      return 'VARCHAR[]';
```

Comment:
nit: Looks like the previous methods return MYSQL values as default but this is default to Postgres.

Reply:
These have been updated with addition of PG tests.